src/main/java/core/basesyntax/serviceimpl/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction: transactions) {
+            FruitTransaction.Operation operation = transaction.getOperation();
+            OperationHandler handler = operationStrategy.getHandler(operation);
+            handler.apply(transaction);
```

Comment:
The 'process' method correctly iterates over the transactions and applies the corresponding operation handlers. This is a good use of the Strategy pattern. Make sure that the 'OperationHandler' is capable of handling errors and edge cases, such as null values or invalid operations.